### ISSUE TITLE
Federated learning not working on multi-layer fully connected model

### ISSUE THREAD

**Issue by mshubhankar** [OpenMined/PySyft#2011]
I am trying to build a federated model with 2 virtual workers bob and alice. I am following code from examples folder to build my model. The model seems to perfectly it is a single fully connected layer but doesn't learning when I increase the layers and change loss function to CrossEntropy. 
**Single Layer:**
**Model -** 
`model = nn.Linear(8,2)
loss= torch.nn.MSELoss()`
**Output -** 
<img width="833" alt="Screenshot 2019-03-25 at 6 56 35 PM" src="https://user-images.githubusercontent.com/11030979/54959572-e5dc7b80-4f2f-11e9-8563-161ba374879a.png">

**Multi Layer :**
**Model -**
 `
class my_network(torch.nn.Module):

     def __init__(self):
        super(my_network, self).__init__()
        self.fc1=nn.Linear(data_width,200)
        self.fc2=nn.Linear(200,100)
        self.fc3=nn.Linear(100,2)
    
    def forward(self,input_):
        a1=F.relu(self.fc1(input_))
        a1=F.relu(self.fc2(a1))
        y=self.fc3(a1)
        return F.softmax(y,dim=1)

     loss= torch.nn.CrossEntropyLoss()
`

 
**Output -** 
<img width="757" alt="Screenshot 2019-03-25 at 7 01 49 PM" src="https://user-images.githubusercontent.com/11030979/54959742-7450fd00-4f30-11e9-80e0-de55728d5b51.png">

This might be a trivial issue but I am new to PyTorch and PySyft and I've been trying to solve this problem for a week without any success.

**Comment by mari-linhares** [comment]
Hey @mshubhankar, I'm having a look at this. I'll let you know if I find anything. Can you share the notebook you're using (a [gist](https://gist.github.com/) link would be great)? Thanks!

**Comment by mshubhankar** [comment]
Thank you for looking into the matter @mari-linhares . The link to the gist code is [here](https://gist.github.com/mshubhankar/94bb81d13290ed29f687fdeafefee5d9) .

**Comment by mari-linhares** [comment]
Hi @mshubhankar, I've modified your code to use a fake dataset and I was able to see the model training using multi-layers and crossentropy ([here](https://gist.github.com/mari-linhares/c6c76ed8933b6cfed785d6c95b34e5fe).

The only big difference from my example to yours is that you're using AdamOptimizer, try to use SGD instead. I think momentum is not yet supported so that's why you may be seeing a strange behavior.

**Comment by mshubhankar** [comment]
Oh thank you so much @mari-linhares !!!! I have been banging my head for this for such a long time now for this!
